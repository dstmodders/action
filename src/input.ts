import * as core from '@actions/core';

export interface Input {
  busted: boolean;
  ignoreFailure: boolean;
  luacheck: boolean;
  prettier: boolean;
  slack: boolean;
  slackLuacheckFormat: string;
  slackPrettierFormat: string;
  slackStyLuaFormat: string;
  stylua: boolean;
}

function getSlackFormat(name: string): string {
  const value = core.getInput(name);
  if (
    value.length > 0 &&
    (value === 'failures' || value === 'issues' || value === 'passes')
  ) {
    return value;
  }
  if (value.length === 0) {
    return 'issues';
  }
  throw new Error(
    `Invalid ${name} input value. Should be: failures|issues|passes`,
  );
}

export async function get(): Promise<Input> {
  try {
    const input: Input = <Input>{};
    input.busted = core.getBooleanInput('busted');
    input.ignoreFailure = core.getBooleanInput('ignore-failure');
    input.luacheck = core.getBooleanInput('luacheck');
    input.prettier = core.getBooleanInput('prettier');
    input.slack = core.getBooleanInput('slack');
    input.slackLuacheckFormat = getSlackFormat('slack-luacheck-format');
    input.slackPrettierFormat = getSlackFormat('slack-prettier-format');
    input.slackStyLuaFormat = getSlackFormat('slack-stylua-format');
    input.stylua = core.getBooleanInput('stylua');
    return input;
  } catch (error) {
    return Promise.reject(error);
  }
}
