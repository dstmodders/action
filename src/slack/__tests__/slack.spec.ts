import * as github from '@actions/github';
import { Slack, SlackOptions } from '../index';
import { Input } from '../../input';

const { expect } = require('expect');

describe('slack', () => {
  const ENV = process.env;
  const GITHUB_CONTEXT = { ...github.context };

  let input: Input;
  let options: SlackOptions;

  beforeAll(() => {
    process.env = { ...ENV };
  });

  afterAll(() => {
    process.env = ENV;
  });

  beforeEach(() => {
    input = <Input>{
      busted: false,
      ignoreCheckVersions: false,
      ignoreFailure: false,
      ignoreSetOutput: false,
      ldoc: false,
      luacheck: false,
      prettier: false,
      slack: false,
      slackForceStatus: '',
      slackLuacheckFormat: '',
      slackPrettierFormat: '',
      slackStyLuaFormat: '',
      stylua: false,
    };

    options = <SlackOptions>{
      channel: 'test',
      signingSecret: '',
      token: '',
      input,
    };
  });

  describe('when initialize Slack class', () => {
    beforeAll(() => {
      github.context.actor = 'user';
      github.context.job = 'Test';
      github.context.runId = 1;
      github.context.workflow = 'CI';
      jest.spyOn(github.context, 'repo', 'get').mockImplementation(() => {
        return {
          owner: 'user',
          repo: 'repository',
        };
      });
    });

    afterAll(() => {
      github.context.actor = GITHUB_CONTEXT.actor;
      github.context.job = GITHUB_CONTEXT.job;
      github.context.runId = GITHUB_CONTEXT.runId;
      github.context.workflow = GITHUB_CONTEXT.workflow;
      jest.restoreAllMocks();
    });

    it('should match the snapshot', async () => {
      expect(new Slack(options)).toMatchSnapshot();
    });
  });
});
