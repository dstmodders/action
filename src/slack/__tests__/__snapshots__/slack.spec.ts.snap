// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`slack when initialize Slack class should match the snapshot 1`] = `
Slack {
  "app": null,
  "channelID": "",
  "isRunning": false,
  "options": {
    "channel": "test",
    "input": {
      "busted": false,
      "ignoreCheckVersions": false,
      "ignoreFailure": false,
      "ignoreSetOutput": false,
      "ldoc": false,
      "luacheck": false,
      "prettier": false,
      "slack": false,
      "slackForceStatus": "",
      "slackLuacheckFormat": "",
      "slackPrettierFormat": "",
      "slackStyLuaFormat": "",
      "stylua": false,
    },
    "signingSecret": "",
    "token": "",
  },
}
`;
